datafusion/expr/src/udf.rs

```diff
@@ -477,6 +487,15 @@ pub trait ScalarUDFImpl: Debug + Send + Sync {
         true
     }
 
+    /// Returns the field metadata for this function.
```

Comment:
I think a more detailed explanation would be useful for someone adding a new UDF to know why this method exists and what use case it is trying to solve. Without looking at this PR it's pretty opaque.